### PR TITLE
[#3247] Support short endpoint names in CoAP adapter

### LINKED ISSUE #3247: CoAP adapter should support short endpoint names
Given that CoAP mainly addresses use cases where constrained devices are involved, it seems reasonable to support short versions of endpoint names in request URIs as well. That is `t` in addition to `telemetry`, `e` in addition to `event` and `r` in addition to `command_response`.

This would save a few bytes with each request message.

### PR DESCRIPTION
**Pull request by sophokles73**
The CoAP adapter now exposes its resources using short endpoint name
alternatives to the existing resources.

Fixes #3247

### CODE REVIEW

**Comment by boaks** [comment]
In the past, the CoAP-option "location" was used for responses carrying a command to inform the client, which URI must be used to report back the command status. maybe, we adapt that as well, e.g. short form in initial request, short form in the location.

**Comment by sophokles73** [comment]
@boaks I have pushed corresponding changes. I guess I covered all of your (excellent) suggestions. Would you mind taking another look?

File: adapters/coap-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java [hunk 1]
```diff
@@ -426,45 +426,73 @@ protected final Future<Void> doUploadMessage(
      * This default implementation adds the command name, content format and response URI to the
      * CoAP response options and puts the command's input data (if any) to the response body.
      *
+     * @param request The CoAP request.
      * @param response The CoAP response.
      * @param commandContext The context containing the command to add.
      * @param currentSpan The Open Tracing span used for tracking the CoAP request.
      */
     protected void addCommandToResponse(
+            final CoapContext request,
```
Comment (boaks) [inline adapters/coap-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java:435]: I'm not that happy to mix the terms `request` and `context`.

**Comment by boaks** [comment]
Beside the term `request` for the `context` (more than the request, carries also some additional state),
it looks good to me.

**Pull request merged**